triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/Request.scala

```diff
@@ -9,8 +9,8 @@ import spray.json.{JsString, JsValue, JsonFormat, deserializationError}
 
 object Request {
   implicit object IdentifierFormat extends JsonFormat[Identifier] {
-    def read(value: JsValue) = value match {
-      case JsString(s) => {
+    def read(value: JsValue): Identifier = value match {
```

Comment:
I could have sworn that there was a warning for this that you could enable and enforce but now I cannot find it anymore :thinking:

Reply:
Yeah, not sure why but IntelliJ's scalafmt highlights more things than scalafmt in our dev env enforces.

Reply:
scalafmt is purely a formatter. IntelliJ can run scalafmt, and we configure it to do so. _Unrelatedly_, IntelliJ has a ton of (mostly useful, sometimes broken) static analysis (which it calls "code inspections") that it can perform. This includes one that tells you to add a return type to any non-private function definition, which is on by default.

We could, in theory, [run IntelliJ code inspections from the command line](https://www.jetbrains.com/help/idea/command-line-code-inspector.html), and therefore in CI, but I've never tried it and I'm pretty sure @S11001001's head would explode if I did.

Reply:
And this is a case where requiring the annotation is silly. It's implied by `read` being an override; `override`, if used, even imposes an expected type context on the body. Moreover, if this was an implicit val, the "public" signature would actually be private.